src/peft/tuners/lora/eva.py

```diff
@@ -43,8 +43,14 @@ class _Hook:
     A base class for hooks that prepares layer inputs for EVA.
     """
 
-    def __init__(self, name: str, prepare_layer_inputs_fn: Optional[callable] = None):
+    def __init__(
+        self,
+        name: str,
+        prepare_layer_inputs_fn: Optional[callable] = None,
+        gather_distributed_inputs: bool = False,
```

Comment:
For my understanding, setting `gather_distributed_inputs=True` in a non-distributed setting makes no sense. In what circumstances would we `gather_distributed_inputs=False` in a distributed setting?

Reply:
My thinking when adding this parameter was that users might use a distributed environment but for some reason do not create a distributed dataloader for EVA. For those cases we would have the same examples as layer inputs and thus when gathering we would have duplicates in the input to the SVD computation. Not sure if this is a valid scenario of if we can disregard this.

Reply:
Ah yes, that makes sense. Could you please add this info to the docstring below, that way users know when they should set this in a distributed setting.

One thing I wonder: Could we make this `Optional[bool]` and if the value is `None` and the setting is distributed, automatically switch to `True`, otherwise to `False`? This is probably the most common case. Users still can pass `False` explicitly in a distributed setting if they want to.

Reply:
I think we can leave it as `bool` and set the default to `True` as the Hooks anyway only gather if `dist.is_initialized()`. So if the setting is not distributed this argument has no effect.